app/BUILD.bazel

```diff
@@ -8,14 +8,43 @@ load("@tools_android//tools/crashlytics:defs.bzl", "crashlytics_android_library"
 load("@tools_android//tools/googleservices:defs.bzl", "google_services_xml")
 load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
 
-# Source files for annotations library.
+'''
+Source files for the annotations library. The files inside the annotations library and the listeners
+library are dependencies for all other libraries in app module.
+
+Place your file here if:
+ - A view, view model, or layout file depends on it
+ - It is an annotation or helper file of some kind
+ - It is not considered a listener
+
+DO NOT place your file here if:
+ - It depends on any other file not included in this list
+ - Any files in the listeners library depends on it
+ - It uses Kotlin synthetic imports
+ - Only test files depend on it
+'''
 ANNOTATIONS = [
     "src/main/java/org/oppia/app/activity/ActivityScope.kt",
     "src/main/java/org/oppia/app/fragment/FragmentScope.kt",
     "src/main/java/org/oppia/app/utility/KeyboardHelper.kt",
 ]
 
-# Source files for listener library.
+'''
+Source files for the listeners library.
+
+The files inside the listeners library and the annotations library are dependencies for all other
```

Comment:
Here & above: it's a bit confusing that you're referring to both lists of files in both locations. Suggest keeping the documentation specific to the thing it's documenting.

Ditto elsewhere.

Reply:
Okay